.github/workflows/test.yml

```diff
@@ -169,3 +169,32 @@ jobs:
           path: |
             test-cache
             ~/test-cache
+
+  test-restore-only:
```

Comment:
I'm not sure this tests the new functionality. Can you elaborate how does this test the restore-only?

In my previous comment, I meant that you should make a new file and use your pattern of "save" and "restore" actions.

Reply:
Note how it uses the new sub-action that restores only via the `uses: ./restore/` here:

https://github.com/mattlewis92/actions-cache/blob/18d4ac58ee9683f4c7ea366fee85bcc0fa578240/.github/workflows/test.yml#L184

They could be put into a different workflow, but it seemed redundant to me as it also functions just fine in this one. I can move it into a separate workflow if you'd prefer though

Reply:
The tests is setup so that it logs the machine ID it runs on and restores on the same machine. I know, it can be so much better, but it works so far. So a separate file will probably put it on a different machine. Hence tests the actual thing. If put in  the same file it'll pick up from the previous step. 

After written all these, I realised you do distinguish using the key. so all good for me.